lib/pubsub/topic.js

```diff
@@ -63,6 +63,27 @@ function Topic(pubsub, options) {
 }
 
 /**
+ * Format a message object as the upstream API expects it.
+ *
+ * @private
+ *
+ * @return {object}
+ */
+Topic.formatMessage_ = function(message) {
+  if (!message.data) {
+    message = {
+      data: new Buffer(message).toString('base64')
+    };
+  }
+
+  if (!util.is(message.data, 'string') && !util.is(message.data, 'buffer')) {
+    message.data = JSON.stringify(message.data);
```

Comment:
It's to publish a message in the API format, which in practicality means you can attach a label to a message. We could easily allow that with `.publish()`, though it would force `publish` to accept an object:

```js
.publish({
  data: "hi",
  label: ["messages"]
})
```